torchvision/prototype/datasets/_builtin/sintel.py

```diff
@@ -0,0 +1,167 @@
+import io
+import pathlib
+import re
+from functools import partial
+from typing import Any, Callable, Dict, List, Optional, Tuple, Iterator, Iterable, TypeVar, Union
+
+import numpy as np
+import torch
+from torchdata.datapipes.iter import (
+    IterDataPipe,
+    Demultiplexer,
+    Mapper,
+    Shuffler,
+    Filter,
+    IterKeyZipper,
+    ZipArchiveReader,
+)
+from torchvision.prototype.datasets.utils import (
+    Dataset,
+    DatasetConfig,
+    DatasetInfo,
+    HttpResource,
+    OnlineResource,
+    DatasetType,
+)
+from torchvision.prototype.datasets.utils._internal import INFINITE_BUFFER_SIZE
+
+T = TypeVar("T")
+
+FILE_NAME_PATTERN = re.compile(r"(frame|image)_(?P<idx>\d+)[.](flo|png)")
+
+try:
+    from itertools import pairwise  # type: ignore
+except ImportError:
+    from itertools import tee
+
+    def pairwise(iterable: Iterable[T]) -> Iterable[Tuple[T, T]]:
+        a, b = tee(iterable)
+        next(b, None)
+        return zip(a, b)
+
+
+class IntCategoryGrouper(IterDataPipe[Tuple[Tuple[str, T], Tuple[str, T]]]):
+    def __init__(self, datapipe: IterDataPipe[Tuple[str, T]]) -> None:
+        self.datapipe = datapipe
+
+    def __iter__(self) -> Iterator[Tuple[Tuple[str, Any], Tuple[str, Any]]]:
+        for item1, item2 in pairwise(sorted(self.datapipe)):
+            if pathlib.Path(item1[0]).parent != pathlib.Path(item2[0]).parent:
+                continue
+
+            yield item1, item2
+
+
+class SINTEL(Dataset):
+    def _make_info(self) -> DatasetInfo:
+        return DatasetInfo(
+            "sintel",
+            type=DatasetType.IMAGE,
+            homepage="",
+            valid_options=dict(
+                split=("train", "test"),
+                pass_=("clean", "final"),
```

Comment:
TODO: we should have `both` as an option here. And this should be renamed to `pass`, however that's a Python keyword and I'll prefer to call it something else.

Reply:
In the other implementation we call it `pass_name`:

https://github.com/pytorch/vision/blob/031e129b5590ec9619962ba44254405fd9ec90c8/torchvision/datasets/_optical_flow.py#L106-L107

```suggestion
                pass_name=("clean", "final"),
```

Reply:
Yes, there is still time to change the original implementation (it hasn't been released yet), so feel free to suggest new names. But otherwise we should try to keep both the current and new version as close as possible, unless there's a really good reason not to